### ISSUE TITLE
mIoU decreases after resuming the checkpoint trained by myself

### ISSUE THREAD

**Issue by huqiming513** [yassouali/pytorch_segmentation#32]
I trained the PSPNet model from scratch for 80 epochs and saved a checkpoint whose training and validating mIoU were about 0.6.
![image](https://user-images.githubusercontent.com/31566437/70310261-2aa00c80-184a-11ea-93a4-e3bd2c1acd3f.png)

After that, I didn't change any setting, but when I resumed from the checkpoint, the training and validating mIoU were just about 0.5. 
![image](https://user-images.githubusercontent.com/31566437/70310358-63d87c80-184a-11ea-91d2-5581de3cc1ca.png)
I have no idea how this could have happened :(
Is it because of the 'lr_scheduler' that commented by you? 
![image](https://user-images.githubusercontent.com/31566437/70310649-1c062500-184b-11ea-82b4-9456a5b5027f.png)
Looking forward to your reply, and thanks very much. :)

**Comment by yassouali** [comment]
Hi, if in the first time you used epochs= 80, at the last iterations the LR  rate was close to zero (`poly`). But if resuming you changed the number of epochs, say to 100, you'll resume with a higher LR which may degrade the performance in the first iterations. I would think, in the end of training, you'll get similar results.

**Comment by huqiming513** [comment]
> Hi, if in the first time you used epochs= 80, at the last iterations the LR rate was close to zero (`poly`). But if resuming you changed the number of epochs, say to 100, you'll resume with a higher LR which may degrade the performance in the first iterations. I would think, in the end of training, you'll get similar results.

Thanks a lot! But why I'll get similar results while I changed the number of epochs to 100 or more?

**Comment by huqiming513** [comment]
Maybe conditions like this? 
![IMG_20191206_183345](https://user-images.githubusercontent.com/31566437/70317338-82457480-1858-11ea-889d-2c6856df25fb.jpg)

**Comment by yassouali** [comment]
It is more like this;

![image](https://user-images.githubusercontent.com/45082182/70318208-ccf6ca80-181f-11ea-9ba4-c7120cc3c134.png)

You have two lines representing the LR for 80 and 100 epochs, if you first train on 80 then resume to train on 100, the LR will be similar to the blue line.

**Comment by huqiming513** [comment]
> It is more like this;
> 
> ![image](https://user-images.githubusercontent.com/45082182/70318208-ccf6ca80-181f-11ea-9ba4-c7120cc3c134.png)
> 
> You have two lines representing the LR for 80 and 100 epochs, if you first train on 80 then resume to train on 100, the LR will be similar to the blue line.

The _results_ you mean are referred to the value of learning rate, but not the model weights or mIoU. Am I right?

**Comment by yassouali** [comment]
What I meant, is that when you suddenly increase the LR (if you resume after 80 epochs), this sudden increase may result in high gradients and reduce the mIoU.

**Comment by huqiming513** [comment]
Ok, thanks a lot. :)
